modules/programs/obsidian.nix

```diff
@@ -0,0 +1,547 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  inherit (lib)
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    literalExpression
+    types
+    ;
+
+  cfg = config.programs.obsidian;
+
+  corePlugins = [
+    "audio-recorder"
+    "backlink"
+    "bookmarks"
+    "canvas"
+    "command-palette"
+    "daily-notes"
+    "editor-status"
+    "file-explorer"
+    "file-recovery"
+    "global-search"
+    "graph"
+    "markdown-importer"
+    "note-composer"
+    "outgoing-link"
+    "outline"
+    "page-preview"
+    "properties"
+    "publish"
+    "random-note"
+    "slash-command"
+    "slides"
+    "switcher"
+    "sync"
+    "tag-pane"
+    "templates"
+    "word-count"
+    "workspaces"
+    "zk-prefixer"
+  ];
+in
+{
+  options.programs.obsidian = {
+    enable = mkEnableOption "obsidian";
+    package = mkPackageOption pkgs "obsidian" { };
+
+    defaultSettings = {
```

Comment:
Are these settings that can be overridden by another mutable config file or something?